docs: remove kibana and dashboards

bmorelli25 opened the pull request:
Removes Dashboards and Kibana from docs per #1673.

**To do after merging:**
 - [ ] Remove unused APM conditionals from the `libbeat/docs/shared-kibana-config.asciidoc` file to make life easier on the beats team

bmorelli25 commented on docs/copied-from-beats/shared-ssl-config.asciidoc:
```diff
@@ -4,7 +4,9 @@
 You can specify SSL options when you configure:
 
 * <<configuring-output,outputs>> that support SSL
+ifeval::["{beatname_lc}"!="apm-server"]
```
This has been added to my Beats PR: https://github.com/elastic/beats/pull/9595/commits/28f90229e60f028619b249c58700983574ba2f6d

bmorelli25 commented on docs/copied-from-beats/outputconfig.asciidoc:
```diff
@@ -219,8 +219,12 @@ The index name to write events to. The default is
 need to configure the `setup.template.name` and `setup.template.pattern` options
 (see <<configuration-template>>).
 
+////
```
Ahh, I was wrong. This line remains, but we've added a `no_dashboards` attribute to hide it. I'll add the attribute the `index.asciidoc` file and commit the changes.

bmorelli25 commented:
Thanks @simitt. I've updated `apm-server.yml`. I also removed a few references to dashboards/kibana in the elasticsearch output section.

graphaelli commented:
@bmorelli25 Can you update apm-server.docker.yml too please?

bmorelli25 commented:
Just realized the `apm-architecture.png` image includes a reference to dashboards that we probably want to remove as well:
<img width="728" alt="screen shot 2019-01-07 at 2 02 08 pm" src="https://user-images.githubusercontent.com/5618806/50795965-d9c41480-1284-11e9-8fa0-6f2395118ba4.png">

Pinging @simitt / @formgeist (because of #198) -- Who created this image / How was it created / How can we update it?

simitt commented:
Actually we still support dashboards, they are just not loaded via APM Server any more, but via Kibana. I think the visualisation is still fine.

formgeist commented:
@bmorelli25 I can update the diagram (and we should because new logos etc.), but it's used on the [APM overview page](https://www.elastic.co/guide/en/apm/get-started/current/overview.html) that covers stuff beyond APM Server. Is it also shown in the APM Server docs?

formgeist commented:
Made some updates to the diagram https://github.com/elastic/apm-server/pull/1739

bmorelli25 commented:
@formgeist - it's only used in the APM Overview. The new image is perfect. Thanks for updating!

bmorelli25 commented:
Rebased with #1634 changes. Added removed links to `redirects.asciidoc`. No longer blocked. OK to merge when you're ready for it @elastic/apm-server team!

simitt commented:
@bmorelli25 I am sorry I overlooked that you made the changes directly in `apm-server.yml`. This is a generated file. You need to apply the changes in https://github.com/elastic/apm-server/blob/master/_meta/beat.yml. Running the command `make update` then generates the `apm-server.yml`. For the make command you need to have a proper python setup. Lmk if you run into any troubles and need help with the setup.

bmorelli25 commented:
@simitt - Doh! Sorry about that. I was able to make the changes in `_meta/beat.yml` and run `make update` to generate the new `apm-server.yml`!
